### DISCUSSION TITLE
How to search for any Landsat TM dataset

### DISCUSSION THREAD

**Discussion by pahbs** [MAAP-Project/Community#1000] [Data] [answered]
We are revisiting how to access Landsat TM collection 2 data across any TM dataset (1984-2024).
This notebook no longer works: https://github.com/developmentseed/example-jupyter-notebooks/blob/landsat-search/notebooks/Landsat8-Search/L8-USGS-satapi.ipynb
Originally, this is how you taught us to access the L8 data to build our composites. When then switched to HLS, so we stopped queries to this api.
Now, we need to:
1. get it working again for Landsat 8
2. learn how to do the same thing for Landsat 5/7
An initial question is how do I learn what api to point to?

**Comment by wildintellect** [comment, marked as answer]
@pahbs 
- https://www.usgs.gov/landsat-missions/landsat-commercial-cloud-data-access
- [Commercial Access User Guide](https://d9-wret.s3.us-west-2.amazonaws.com/assets/palladium/production/s3fs-public/atoms/files/LSDS-2032-Landsat-Commercial-Cloud-Direct-Access-Users-Guide-v2.pdf.pdf)
- https://www.usgs.gov/landsat-missions/spatiotemporal-asset-catalog-stac

Looks like they changed the server url:
```
from pystac_client import Client
catalog = "https://landsatlook.usgs.gov/stac-server"
cat = Client.open(catalog)
for collection in cat.get_all_collections():
    print(collection)
#... take the collection name and use below

# Filter by platform
query = {
    "platform": {"eq": "LANDSAT_5"}
}


# can also add bbox, and datetime ranges here
results = cat.search(
    collections='landsat-c2l2-sr',
    max_items=10,
    query=query
)

for item in results.items():
    print(item.id)

```
```
LT05_L2SP_038037_20120505_20200820_02_T1_SR
LT05_L2SP_031036_20120504_20200820_02_T1_SR
LT05_L2SP_031035_20120504_20200820_02_T1_SR
LT05_L2SP_031034_20120504_20200820_02_T1_SR
LT05_L2SP_031033_20120504_20200820_02_T1_SR
LT05_L2SP_039038_20120426_20200820_02_T1_SR
LT05_L2SP_039037_20120426_20200820_02_T1_SR
LT05_L2SP_039036_20120426_20200820_02_T1_SR
LT05_L2SP_039035_20120426_20200820_02_T1_SR
LT05_L2SP_039034_20120426_20200820_02_T1_SR
```
I would also recommend updating the code to use `pystac_client` instead of hand writing the STAC API calls and return handling.

**Reply by pahbs** [reply]
@wildintellect this is great - so if I were to write a script to access & process these data to match the way we work with HLS (https://github.com/lauraduncanson/icesat2_boreal/blob/3086c7b529fb3fac5a71f01dfd2daa21977c2d70/lib/fetch_HLS.py#L73) would there be any differences associated with AWS access? 

note that use `pystac_client` to fetch HLS and would update that script to work with all Landsat TM based on what you have above.

**Reply by wildintellect** [reply]
I believe these products would fall under a `request-pays` bucket so you'd need to add that to the actual read calls made with rasterio. Pretty sure you have a previous example.
